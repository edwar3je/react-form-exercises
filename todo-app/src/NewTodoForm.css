.todo-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 1em;
}

.submit-button {
    height: 30px;
    width: 150px;
    font-size: 15px;
    font-weight: bold;
    color: white;
    background-color: rgb(45, 196, 45);
}