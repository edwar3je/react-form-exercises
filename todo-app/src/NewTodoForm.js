import './NewTodoForm.css';
import { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NewTodoForm = ({ add }) => {

    const initialState = {
        task: ""
    };

    const input = document.querySelector('input.task');

    const resetValue = () => {
        input.value = ""
    };
    
    const [todoTask, setTodoTask] = useState(initialState);
    
    const handleChange = (e) => {
        setTodoTask({ 
            task: e.target.value,
            id: uuidv4()
        })
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        const { task, id } = todoTask;
        console.log(id);
        add(task, id);
        alert(`Created new todo task: ${task}`);
        setTodoTask(initialState);
        resetValue();  
    };
    
    return (
        <form className="todo-form" onSubmit={handleSubmit}>
            <label htmlFor="task">Task</label>
            <input type="task" className="task" id="task" placeholder="todo" onChange={handleChange}></input>
            <button type="submit" className="submit-button">Add New Task</button>
        </form>
    );
}

export default NewTodoForm;