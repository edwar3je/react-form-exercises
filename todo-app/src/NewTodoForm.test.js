import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("renders without crashing", () => {
    render(
        <NewTodoForm 
            add={() => console.log("you added a new box")}
        />
    );
});

it("matches the snapshot", () => {
    const { asFragment } = render(
        <NewTodoForm 
            add={() => console.log("you added a new box")}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});