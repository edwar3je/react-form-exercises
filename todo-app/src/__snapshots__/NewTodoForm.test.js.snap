// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="todo-form"
  >
    <label
      for="task"
    >
      Task
    </label>
    <input
      class="task"
      id="task"
      placeholder="todo"
      type="task"
    />
    <button
      class="submit-button"
      type="submit"
    >
      Add New Task
    </button>
  </form>
</DocumentFragment>
`;
